### PR TITLE
Add .NET Native builds for Dynamic.Runtime, consolidate projects

### PR DESCRIPTION
**Pull request by mellinoe**
This adds the .NET Native (and "NetCoreForCoreCLR") builds for System.Dynamic.Runtime. I've also consolidated down the net46 build into the single project file in the src directory. The configurations here are a bit more complicated than usual.

@weshaggard

### CODE REVIEW

File: src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj [hunk 1]
```diff
@@ -6,14 +6,28 @@
...
     <AssemblyVersion>4.0.11.0</AssemblyVersion>
-    <DefineConstants Condition=" '$(IsInterpreting)' != 'true' ">$(DefineConstants);FEATURE_COMPILER</DefineConstants>
+    <IsInterpreting Condition="'$(TargetGroup)' == 'netcore50' or '$(TargetGroup)' == 'netcore50aot'">true</IsInterpreting>
+    <DefineConstants Condition="'$(IsInterpreting)' != 'true'">$(DefineConstants);FEATURE_COMPILER</DefineConstants>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
   </PropertyGroup>
+
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' " />
```
Comment (weshaggard) [inline src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj]: nit: kill spaces at the beginning and end of these conditions.

File: src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj [hunk 2]
```diff
@@ -69,7 +83,7 @@
       <Link>Common\System\Runtime\CompilerServices\TrueReadOnlyCollection.cs</Link>
     </Compile>
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true'">
```
Comment (weshaggard) [inline src/System.Dynamic.Runtime/src/System.Dynamic.Runtime.csproj]: probably worth merging these 2 ItemGroups to elimate the one condition.

**Comment by weshaggard** [comment]
cc @VSadov

**Comment by VSadov** [comment]
LGTM

**Comment by weshaggard** [comment]
LGTM

**Comment by mellinoe** [comment]
@mmitche I think the machine is out of space here:

```
14:44:52 FATAL: Unable to produce a script file
14:44:52 java.io.IOException: Failed to create a temp file on /mnt/j/workspace/dotnet_corefx/ubuntu_debug_tst_prtest@2
Caused by: java.io.IOException: remote file operation failed: /mnt/j/workspace/dotnet_corefx/ubuntu_debug_tst_prtest@2 at hudson.remoting.Channel@fd5fe56:dci-ub-fbld-2: java.io.IOException: No space left on device
```

Will restarting that run fix this, or do we need manual intervention?

**Comment by mellinoe** [comment]
@dotnet-bot Test Innerloop Ubuntu Debug Build and Test

**Pull request merged**